sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py

```diff
@@ -120,19 +124,45 @@ class ServiceBusReceiver(
      (if provided) within its request to the service.
     """
 
-    def __init__(self, fully_qualified_namespace, credential, **kwargs):
-        # type: (str, Union[TokenCredential, AzureSasCredential, AzureNamedKeyCredential], Any) -> None
+    def __init__(
+        self,
+        fully_qualified_namespace: str,
+        credential: Union["TokenCredential", "AzureSasCredential", "AzureNamedKeyCredential"],
+        *,
+        queue_name: Optional[str] = None,
+        topic_name: Optional[str] = None,
+        subscription_name: Optional[str] = None,
+        http_proxy: Optional[Dict[str, Any]] = None,
+        user_agent: Optional[str] = None,
+        logging_enable: Optional[bool] = False,
+        transport_type: Optional[TransportType] = TransportType.Amqp,
+        receive_mode: Union[
+            ServiceBusReceiveMode, str
+        ] = ServiceBusReceiveMode.PEEK_LOCK,
+        max_wait_time: Optional[float] = None,
+        auto_lock_renewer: Optional["AutoLockRenewer"] = None,
+        prefetch_count: int = 0,
```

Comment:
I would not expose `http_proxy`, `logging_enable`, `transport_type` - I'd just leave these in kwargs + docs.